sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -198,14 +257,46 @@ class StreamExecution(
     logDebug(s"Unblocked at $newOffset for $source")
   }
 
+  override def awaitTermination(): Unit = {
+    require(state != INITIALIZED,
```

Comment:
Right! My bad. But what should be the exception in this case. This can be an assert() though I dont think this can be thrown by any user action, as the user will always get back a query in ACTIVE state.